README.md

```diff
@@ -659,10 +659,290 @@ To begin coding the basic execution functionality of our VM, we only need these
+
+```go
+func (run_context RunContext) ComputeDstAddr(instruction Instruction) (memory.Relocatable, error) {
+	var base_addr memory.Relocatable
+	switch instruction.DstReg {
+	case AP:
+		base_addr = run_context.Ap
+	case FP:
+		base_addr = run_context.Fp
+	}
+
+	if instruction.Off0 < 0 {
+		return base_addr.SubUint(uint(math.Abs(float64(instruction.Off0))))
+	} else {
+		return base_addr.AddUint(uint(instruction.Off0))
+	}
+
+}
+```
+##### ComputeOp0Addr
+
+The process is similar to compute the dst address.
+
+```go
+func (run_context RunContext) ComputeOp0Addr(instruction Instruction) (memory.Relocatable, error) {
+	var base_addr memory.Relocatable
+	switch instruction.Op0Reg {
+	case AP:
+		base_addr = run_context.Ap
+	case FP:
+		base_addr = run_context.Fp
+	}
+
+	if instruction.Off1 < 0 {
+		return base_addr.SubUint(uint(math.Abs(float64(instruction.Off1))))
+	} else {
+		return base_addr.AddUint(uint(instruction.Off1))
+	}
+}
+
+```
+
+##### ComputeOp1Addr
+
+It computes the address of `Op1` based on  the `Op0` operand and the kind of Address the instruction has for `Op1`, If its address is `Op1SrcFp` it calculates the direction from Fp register, if it is `Op1SrcAp` then if calculates it if from Ap register. If it is an immediate then checks if the offset 2 is 1 and calculates it from the `Pc`. Finally if its an `Op1SrcOp0` it checks the `Op0` and calculates the direction from it. Then if performs and addition or a substraction if the `Off2` is negative or positive.
+
+```go
+func (run_context RunContext) ComputeOp1Addr(instruction Instruction, op0 *memory.MaybeRelocatable) (memory.Relocatable, error) {
+	var base_addr memory.Relocatable
+
+	switch instruction.Op1Addr {
+	case Op1SrcFP:
+		base_addr = run_context.Fp
+	case Op1SrcAP:
+		base_addr = run_context.Ap
+	case Op1SrcImm:
+		if instruction.Off2 == 1 {
+			base_addr = run_context.Pc
+		} else {
+			base_addr = memory.NewRelocatable(0, 0)
+			return memory.Relocatable{}, &VirtualMachineError{Msg: "UnknownOp0"}
+		}
+	case Op1SrcOp0:
+		if op0 == nil {
+			return memory.Relocatable{}, errors.New("Unknown Op0")
+		}
+		rel, is_rel := op0.GetRelocatable()
+		if is_rel {
+			base_addr = rel
+		} else {
+			return memory.Relocatable{}, errors.New("AddressNotRelocatable")
+		}
+	}
+
+	if instruction.Off2 < 0 {
+		return base_addr.SubUint(uint(math.Abs(float64(instruction.Off2))))
+	} else {
+		return base_addr.AddUint(uint(instruction.Off2))
+	}
+}
+```
+
+##### DeduceOp0
+
+The method deduces the value of `Op0` if possible (based on `dst` and `Op1`).
+If Instruction opcode is a call `Op0` is deduced from the program counter and added the instruction size.
+If it is and `AssertEq` then a second switch case is used to check what `ResLogic` is, if it is `ResAdd` `Op0` is deduced from the substraction of `Op1` to `Dst` or if is is `Resmul` the `Op0` is deduced from the division of `Dst` and `Op1` (both felt values). otherwise op0 is nil.  
```

Comment:
```suggestion
If it is an `AssertEq` then a second switch case is used to check what `ResLogic` is, if it is `ResAdd` `Op0` is deduced from the substraction of `Op1` to `Dst` or if is is `Resmul` the `Op0` is deduced from the division of `Dst` and `Op1` (both felt values). otherwise op0 is nil.  
```